### PR TITLE
Readme Installation improvements. Fixes Automattic/Picard#8

### PR DESCRIPTION
**Pull request by artpi**
I added detailed explanation for installation process, configuration files via localize_script and router changes to accomodate folder installs.
Fixes issues #8 and #22 .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
-1. Install and activate the [WP REST API](https://github.com/WP-API/WP-API/tree/master) plugin (make sure that you are using the `master` branch – the default is currently develop).
-2. Clone this repo to your computer in the `wp-content/themes` directory of your WordPress installation (e.g. `git clone git@github.com:Automattic/Picard.git`).
-3. Run `npm install` to install the node dependencies.
-4. Run `gulp build` to compile the JavaScript and Sass.
-6. Run `gulp watch` to actively develop the theme and have the styles and JS files automatically update.
-7. Set your permlink structure to `/%year%/%monthnum%/%day%/%postname%/`.
+### 1. Working Wordpress installation
+First, prepare proper Wordpress enviroment:
+
+* You will obviously need working *Wordpress* installation,
```
Comment (kopepasah) [inline README.md]: @artpi make sure you are capitalizing the `P` in WordPress. There are several instances mistake of this in your PR.

File: README.md [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
+* You will also need *WP REST API*. It is a plugin (with plans to incorporate into Wordpress core) that creates REST API to be used by theme. Plugin is currently **NOT** in plugins repository, you have to upload the files manually. Install and activate the [WP REST API](https://github.com/WP-API/WP-API/tree/master) plugin (make sure that you are using the `master` branch – the default is currently develop),
+* Set your permlink structure to `/%year%/%monthnum%/%day%/%postname%/`.
+
+### 2. Theme building (for _team_ building go to your local meetup)
+Unlike other themes, this one uses build process. JavaScript is an interpreted language, but we need to take certain steps (like transpiling React JSX syntax or SASS CSS syntax) to take the files from development phase to production. You will need following tools:
+
+* *node.js* and *npm* - node is an command line Javascript interpretter that is used for build process. Npm stands for node package manager. You can install both from [nodejs.org](https://nodejs.org/download/),
```
Comment (kopepasah) [inline README.md]: @artpi same goes for JavaScript.

Comment (kopepasah) [inline README.md]: @artpi since the official site for `npm` does not capitalize the acronym, I would not capitalize the `n` here.

Comment (artpi) [inline README.md]: @kopepasah, thanks for pointing out the typos! Fixed it already.
I`m feeling a little bit ashamed :).

**Comment by artpi** [comment]
@kopepasah, I moved changes relating to this issue to another branch and made pull request. Is it ok? Can I make another pull request relating to issue #22 ?

File: README.md [hunk 3]
```diff
@@ -1,17 +1,32 @@
...
-2. Clone this repo to your computer in the `wp-content/themes` directory of your WordPress installation (e.g. `git clone git@github.com:Automattic/Picard.git`).
-3. Run `npm install` to install the node dependencies.
-4. Run `gulp build` to compile the JavaScript and Sass.
-6. Run `gulp watch` to actively develop the theme and have the styles and JS files automatically update.
-7. Set your permlink structure to `/%year%/%monthnum%/%day%/%postname%/`.
+### 1. Working WordPress installation
+First, prepare proper WordPress enviroment:
+
+* You will obviously need working *WordPress* installation,
+* You will also need *WP REST API*. It is a plugin (with plans to incorporate into WordPress core) that creates REST API to be used by theme. Plugin is currently **NOT** in plugins repository, you have to upload the files manually. Install and activate the [WP REST API](https://github.com/WP-API/WP-API/tree/master) plugin (make sure that you are using the `master` branch – the default is currently develop),
```
Comment (kopepasah) [inline README.md]: @artpi I may be speaking as a developer, but this is a little overly descriptive. I am sure that most individuals using Picard understand the concept of plugins, as Picard is merely an experimental theme.

**Comment by kopepasah** [comment]
@artpi overall, I think may of these changes are targeted for a non-developer and not necessary. If a user needs to be told to setup a WordPress environment, most likely he/she will not be using this experimental theme (much like they will not use [_s](https://github.com/Automattic/_s/).

You addressed many things here, but is there any particular steps in the current README.md that you feel are misleading or need more information?

**Comment by artpi** [comment]
My reasons for writing a detailed README are as follows:
* There is a issue #8 by @jacklenox  with a description: `We need more thorough explanations of setting up node, gulp etc.`
* As you said, this is an experimental theme. So it contains several new concepts for some WordPress developers. A lot of folks out there are not familiar with node, npm, gulp, browserify, theme building, sass, etc..., because they are used to developing themes in PHP. If it is a showcase and a biolerplate, people have to understand what's going on.
* The same @jacklenox in his [WordCamp London talk](http://jacklenox.com/2015/03/30/building-themes-with-the-wp-rest-api-wordcamp-london-march-2015/) mentioned several times, that he will add more verbose README file to the project. Since then, he did not find the time to do so and I decided to help,
* During WordCamp Europe in the end of June he will give a talk titled `Theming, React and the REST API`. I assume, that he will talk about Picard to an audience of several hundred people, and I think it would be best to have a good README,
* What is the downside?

And, answering your question:
The particular step in current README that is misleading is gulp. Following these steps I will not have it in my path, so I can't use it in command line.

**Comment by kopepasah** [comment]
@artpi all valid points and you are correct, there is no downside. I will keep my eye out and provide feedback to your changes as you go.

**Comment by jacklenox** [comment]
Based on the two new PRs, I'll close this one if there are no objections.

**Comment by artpi** [comment]
Sure thing, this one is to be closed.

**Pull request closed without merging**